### DISCUSSION TITLE
Does FailoverClient support marking a read command on master?

### DISCUSSION THREAD

**Discussion by changsongl** [redis/go-redis#1894] [Q&A]
For some scenario, we would like to read on the master node without master-slave lag. Does FailoverClient support it? It's any reason that go-redis not support this feature?

**Comment by vmihailenco** [comment]
You can try `NewFailoverClusterClient` with `Options.RouteRandomly`.

**Reply by changsongl** [reply]
> RouteRandomly

@vmihailenco Thank you for the reply. I may not describe my question clear enough. I wish I could do command and route by myself. 

For example:
````golang
client.Master().DoSomeCMD(...);
client.Slave().DoSomeCMD(...);
````

I read the code of go-redis failover client, and I found out it is only support slave only or master only.

````golang
if failover.opt.SlaveOnly {
	addr, err = failover.RandomSlaveAddr(ctx)
} else {
	addr, err = failover.MasterAddr(ctx)
	if err == nil {
         	failover.trySwitchMaster(ctx, addr)
	}
}
````

Should I use two client? One is for master only, and the other one is for slave only. Then I can route by myself. Like read command on master node.

**Reply by baylac** [reply]
that's what we do, we have our wrapper around redis.UniversalClient, like this:
```
type RedisClient interface {
	Master() redis.UniversalClient
	Replica() redis.UniversalClient
}
```
